CMakeVOL.cmake

```diff
@@ -30,9 +30,9 @@ endfunction ()
 option (HDF5_VOL_ALLOW_EXTERNAL "Allow building of external HDF5 VOL connectors with FetchContent" OFF)
 mark_as_advanced (HDF5_VOL_ALLOW_EXTERNAL)
 if (HDF5_VOL_ALLOW_EXTERNAL)
-  if (HDF5_ALLOW_EXTERNAL_SUPPORT MATCHES "NO" OR NOT HDF5_ALLOW_EXTERNAL_SUPPORT MATCHES "GIT")
-    message (FATAL_ERROR "HDF5_ALLOW_EXTERNAL_SUPPORT must be set to 'GIT' to allow building of external HDF5 VOL connectors")
-  endif ()
+  if (HDF5_ALLOW_EXTERNAL_SUPPORT MATCHES "NO" OR (NOT HDF5_ALLOW_EXTERNAL_SUPPORT MATCHES "GIT" AND NOT HDF5_ALLOW_EXTERNAL_SUPPORT MATCHES "LOCAL_DIR"))
```

Comment:
LOCAL_DIR is not currently a valid choice - just TGZ

Reply:
For maintenance reasons, I didn't want to create a separate path just for support for VOL connectors, since external VFD plugins are another thing we'll have to think about at some point. There shouldn't be any need to grab ZLIB and SZIP from github just by turning on `HDF5_ALLOW_EXTERNAL_SUPPORT` though. I think that's just because the options for building SZIP and ZLIB are on by default.

Reply:
HDF5_ALLOW_EXTERNAL_SUPPORT is just three states NO: go find it, GIT: use a repo and TGZ: use a file.
Because at that time it was all do the same thing, compression, plugins, etc.
There is a subcategory per item: XXX_USE_EXTERNAL which determines whether to do a find_package or use the HDF5_ALLOW_EXTERNAL_SUPPORT option.  See the CMakeFilters.cmake file.
For the vol you could create a VOL_USE_EXTERNAL or even a new variable and leave the last option to follow the global HDF5_ALLOW_EXTERNAL_SUPPORT. 
Does this need refactored - yes, but maybe for version 2.0 of HDF5.

Reply:
There are a couple lines in `CMakeFilters.cmake` that tell CMake that ZLIB will come from github if "GIT" is set:
```
if (HDF5_ALLOW_EXTERNAL_SUPPORT MATCHES "GIT" OR HDF5_ALLOW_EXTERNAL_SUPPORT MATCHES "TGZ")
  set (ZLIB_USE_EXTERNAL "Use External Library Building for ZLIB" 1)
```

You can get around this by disabling ZLIB and SZIP when getting a VOL from Github, or just getting them from Github as well, but it's an unintuitive thing to encounter.